entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -411,17 +412,18 @@ class EntityService(
             .map { getFullEntityById(it) }
     }
 
+    // TODO improve validation and multi attribute support for relationships and geoproperties
     @Transactional
     fun appendEntityAttributes(
         entityId: String,
         attributes: Map<String, Any>,
         disallowOverwrite: Boolean
     ): UpdateResult {
-        val updateStatuses = attributes.map {
-            val attributeValue = expandValueAsMap(it.value)
-            if (!attributeValue.containsKey("@type"))
+        val updateStatuses = attributes.flatMap {
+            val attributeValue = expandValueAsListOfMap(it.value)
+            if (!attributeValue[0].containsKey("@type"))
                 throw BadRequestDataException("@type not found in $attributeValue")
-            val attributeType = attributeValue["@type"]!![0]
+            val attributeType = attributeValue[0]["@type"]!![0]
             logger.debug("Fragment is of type $attributeType")
             if (attributeType == NGSILD_RELATIONSHIP_TYPE.uri) {

```

Comment:
for a relationship, you only append the first relationship and not the others (as you do for the properties)?

Reply:
Yes

Reply:
add it in the talk list of the issue

Reply:
done